packages/ButtonGroup/src/ButtonGroup.js

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+
+import * as sc from "./ButtonGroup.styles";
+import ButtonItem from "./components/ButtonItem";
+
+const propTypes = {
+  /** The content of each of the toggle buttons in the group. */
+  children: PropTypes.node,
+
+  /** To show or hide the icons used for selected or not selected. */
+  hasIcons: PropTypes.bool,
+
+  /** If the button is disabled. */
+  isDisabled: PropTypes.bool,
+
+  /** If the width of the button should span it's parent container (100%). */
+  isFullWidth: PropTypes.bool,
+
+  /** If it will be rendered as a <button> element. If false, a <span> will be rendered via an accessible <RawButton>. */
+  isSemantic: PropTypes.bool,
+
+  /** Callback to be executed when any button is clicked to select the item or activated by keyboard. Typically required.
+   * it will return a list of selected items */
+  onChange: PropTypes.func,
+
+  /** Size of the button (font size, min-height, padding, etc). */
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  children: null,
+  hasIcons: false,
+  isDisabled: false,
+  isFullWidth: false,
+  isSemantic: true,
+  onChange: () => {},
+  size: ShirtSizes.MEDIUM,
+};
+
+const ButtonGroup = props => {
+  const { children, hasIcons, isDisabled, isFullWidth, isSemantic, onChange, size } = props;
+
+  const validChildren = children.filter(child => child.type.displayName === ButtonItem.displayName);
```

Comment:
I think this way of extracting children will fail in different user scenarios as:
```jsx
function renderButtonItems()
  return (<React.Fragment>
     < ButtonGroup.Item>1< ButtonGroup.Item>
     < ButtonGroup.Item>2< ButtonGroup.Item>);
</React.Fragment>)
```

You can use extractChildren function from the helpers and you should have better coverage. 

https://github.com/acl-services/paprika/blob/master/packages/helpers/src/extractChildren.js

how to use it:
https://github.com/acl-services/paprika/blob/master/packages/SidePanel/src/SidePanel.js#L107

Reply:
I can see this has already been refactored some on the way from ACL UI so...
I still don't have 100% 🦅 👁️on this, but I'm ~90% sure that we're again iterating over children when it's not necessary. Seems like everything here can be achieved with context. @Nandez89 an example of what I mean: https://github.com/acl-services/paprika/pull/134

Reply:
^ yeap context can also be used here. thought its a small component with not too many variations.

Reply:
Well it was you who brought up the fragment thing 😄 that won’t work with `extractChildren` but it will work with context. Just in general the more people are aware of this the better

Reply:
These are both interesting cases: 
On the extractChildren approach would be good to use and I its more explicit code for readability, and avoiding having to manually loop over the child seems like a good trade off too... 
I still don't have too much practice with the use of the Context though... so I find it a bit obscure when I'm trying to understand how the context will filter only certain allowed children to be passed in. 

I will look into it more and if I get caught in a loop (🥁) I'll hand it over to Mike to help me out

Reply:
Hey @Nandez89 take a look at this I wrote it after this threat :) the option number two it's the one that talks about these specific cases [SubComponent Pattern](https://github.com/acl-services/paprika/wiki/React-SubComponent-Pattern#2-composing-your-parent-component-with-different-subcomponents) there its a codesandbox example at the bottom so you can see the code working. 

in short for this case is better to use Context, I take back my initial suggestion, give it a try if you find it hard, we can go with this solution and refactor it later :)

Reply:
I'm not sure what the utility is in filtering children in this case. If you stick something other than a buttongroup item into a buttongroup then... you should know you're not gonna have something that makes sense or renders well? 🤷‍♂️